### ISSUE TITLE
adding `isNewSerializerAPI: true` sets embedded relational data JsonAPI format

### ISSUE THREAD

**Issue by koryteg** [adopted-ember-addons/active-model-adapter#76]
I was getting an error in my tests that said need to set `isNewSerializerAPI` to `true` on my serializer.
when I do it changes the data format from what should be this:

``` json
{"name":"name one",
"sent_at":"Wed Jan 06 2016 17:49:47 GMT-0800 (PST)",
"template":"template1",
"filter": { 
            "has-donated":"yes",
            "any-all":"all"
    }              
  }
}
```

to this:

``` json
{"name":"name one",
"sent_at":"Wed Jan 06 2016 17:49:47 GMT-0800 (PST)",
"template":"template1",
"filter": {"data":{
             "attributes":{"has-donated":"yes",
                                 "any-all":"all"
             },
             "relationships":{"campaign-email":
                {"data":{"type":"campaign-emails","id":null}}},"type":"filters"
             }
   }
}
```

this is my serializer:

``` javascript
import DS from 'ember-data';
import ApplicationSerializer from './application';
export default ApplicationSerializer.extend(DS.EmbeddedRecordsMixin, {
  attrs: {
    users: {
      serialize: 'id'
    },
    filter: {
      embedded: 'always'
    }
  }
});
```

**Comment by fivetanley** [comment]
What version of Ember Data, Ember, and Active model Adapter are you using?
What code is calling this?
Can you reproduce in a jsbin at all, and/or post your model definitions?

**Comment by koryteg** [comment]
Active-model-adapter  : 1.13.6
Ember                          : 1.13.11
Ember Data                 : 1.13.15

ill try and get the rest to you soon.

**Comment by fivetanley** [comment]
I would suggest trying the latest version of ActiveModelAdapter which is currently in the 2.x series. I think it is still compatible with 1.13.

**Comment by koryteg** [comment]
I got the same error when updating to AMS `2.0.3` and reproduced it with a sample app here: https://github.com/koryteg/ams-sample

**Comment by fivetanley** [comment]
@koryteg Sorry for the delay :(. Thanks for adding a test case though, it was extremely helpful in debugging.

## TLDR; Add an ApplicationAdapter and `import '../test-helper'` in your unit tests as written below.

Ok, this was a bit of a tricky one to track down.

## The bug

In the embedded records mixin, we call `snapshot.serialize`, this does a few things to get the JSON:

*looks up the model's adapter using `store.adapterFor`, then serializes the model using the found adapter's `defaultSerializer`

The lookup order is something like `modelName`, `'application'`, `default for store (JSONAPI)`. I can't remember the exact order but it's in the store.js source code.

So, the workaround for now is to add an application adapter:

```javascript
//app/adapters/application.js

import ActiveModelAdapter from 'active-model-adapter';

export default ActiveModelAdapter.extend({

});
```

Also, due to the way tests work, you'll need to import `test-helper` so the resolver is set correctly in unit tests. I'm not sure why that's not a default in ember-cli. Anyway:

```javascript
// tests/unit/serializers/comment.js

import '../test-helper';

moduleFor('serializers:comment');

// test code...
```

**Comment by fivetanley** [comment]
On our side we should:

* Generate an application adapter for you
* Throw an exception if you don't have one define, and instruct you to do so.

**Comment by koryteg** [comment]
thanks for hunting this down. it was really throwing me for a loop.

**Comment by jherdman** [comment]
Closing due to inactivity. Please feel free to reopen if this issue is still presenting itself.
